### PR TITLE
api-on-the-fly - part 3 - cylc types

### PR DESCRIPTION
**Pull request by oliver-sanders**
Addresses the "support for custom Cylc types" point in #339 

**Note** - *Built atop #359*
**Note** - *Works with the data types defined in cylc/cylc-flow#3469*

**Overview:**

Configure Vuetify form components to work with the Cylc data types defined in cylc/cylc-flow#3469.

* Generate a sample mutation which has one argument for each data type present in the GraphQL schema for testing and evaluation.
* Mask input values (e.g. ISO8601 date-times).
* Add informative placeholders where appropriate.
* Add basic validation where possible.
* Support composite types (AKA input objects).
* Improve the list component.
* Fix the model persistence bug in `FormGenerator` (so the component can be reused).
* Make the first steps towards separating Vuetify from the form generator.
* Bring in documentation for arguments from the schema (render it as markdown).

**Unrelated Changes:**

* Upgrade Vuetify (older version missing `flat` and has some bugs with field outlines)

**What You Should Expect To See:**

* Forms should be slicker.
* No traceback (all default values should appear).
* All composite (input) types should be working (e.g. `BroadcastSettings`).
* It's not going to be perfect yet but should be more than sufficient for beta release.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [ ] Does contain off-topic changes (listed above).
- [x] Appropriate tests are included (unit and/or functional).
- [x] No change log entry required (still a dummy view not for user consumption)
- [x] No documentation update required.

### CODE REVIEW

**Comment by kinow** [comment]
Sorry for the delay on reviewing it. Tomorrow will keep working on the infinite tree, but will try to review either in the afternoon break after lunch, or by Friday :+1:

File: src/components/graphqlFormGenerator/FormInput.vue [hunk 1]
```diff
@@ -110,12 +104,12 @@ export default {
       // const ofType = this.gqlType.ofType
 
       var componentProps
```
Comment (hjoliver) [inline src/components/graphqlFormGenerator/FormInput.vue]:
```suggestion
      let componentProps
```
 (but ... not actually a change on this branch)

**Comment by hjoliver** [comment]
Minor conflicts

**Comment by hjoliver** [comment]
(Fixed conflicts).

**Pull request merged**